DisruptorUnity3d/Assets/RingBuffer.cs

```diff
@@ -47,13 +47,13 @@ public int Capacity
         /// <returns>The next available item</returns>
         public T Dequeue()
         {
-            var next = _consumerCursor.ReadFullFence() + 1;
-            while (_producerCursor.ReadFullFence() < next)
+            var next = _consumerCursor.ReadAcquireFence() + 1;
```

Comment:
Read acquire should be used in order to make sure that we read the array value **after** we have read the producer cursor. Otherwise, we could read the array to soon and get an older value.